compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/NemoOptimizer.java

```diff
@@ -113,6 +123,23 @@ public IRDAG optimizeAtRunTime(final IRDAG dag, final Message message) {
     return optimizationPolicy.runRunTimeOptimizations(dag, message);
   }
 
+  /**
+   * Operations to be done prior to the Compile-Time Optimizations.
+   * @param dag the DAG to process.
+   * @param policy the optimization policy to optimize the DAG with.
+   */
+  private void beforeCompileTimeOptimization(final IRDAG dag, final Policy policy) {
+    if (policy.getClass().getName().contains("XGBoost")) {
```

Comment:
Can't we use `instanceof` method? encoding strings in the code is not good because we should manually change the name whenever we modify the class name or change the class.

Reply:
You're right. I'll refactor code. Thanks!